packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -822,6 +822,16 @@ export interface DatabaseInstanceSourceProps extends DatabaseInstanceNewProps {
    * @default - no name
    */
   readonly databaseName?: string;
+
+  /**
+   * The parameters in the DBParameterGroup to create automatically
+   *
+   * You can only specify parameterGroup or parameters but not both.
+   * You need to use a versioned engine to auto-generate a DBParameterGroup
+   *
+   * @default - None
+   */
+  readonly parameters?: { [key: string]: string };
```

Comment:
Curious why are we doing this in `DatabaseInstanceSourceProps`, and not in `DatabaseInstanceNewProps`, which is where `parameterGroup` is defined?

Reply:
The DatbaseInstanceNew does not know the engine yet, so we cannot create a ParameterGroup.

Reply:
Makes sense.